src/deadlock-detection/lockorder.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2019 Greg Griffith
+// Copyright (c) 2019 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "lockorder.h"
+#include "util.h"
+
+#ifdef DEBUG_LOCKORDER // this ifdef covers the rest of the file
+
+void CLockOrderTracker::potential_lock_order_issue_detected(const CLockLocation &thisLock,
+    const CLockLocation &otherLock,
+    const uint64_t &tid)
+{
+    LOGA("POTENTIAL LOCK ORDER ISSUE DETECTED\n");
+    LOGA("This occurred while trying to lock: %s after %s \n", thisLock.GetMutexName().c_str(),
+        otherLock.GetMutexName().c_str());
```

Comment:
Fun fact:  %s accepts std::string in LOG/LOGA because it uses tinyformat.h format.  So you don't need .c_str()